### PR TITLE
Load test proc mount in a workspace

### LINKED ISSUE #8098: [integration-test] Load test proc mount in a workspace
## Is your feature request related to a problem? Please describe
On several occasions we have broken Docker containers in a workspace due to intermittently failing proc mounts, e.g. due to libseccomp issues (see #7793). We've seen reports that the problem seems to persist https://github.com/gitpod-io/gitpod/issues/8070.

We must add an integration test which runs the proc stress test (mount `/proc` in a mount namespace) to detect this issue before bringing it to prod.

### PR DESCRIPTION
**Pull request by utam0k**
## Description

* Add stress test for mount proc
* Fix workspace tests not running

## Related Issue(s)

Fixes #8098 

## Release Notes

```release-note
Add stress test for mount proc
```

## Documentation

No

### CODE REVIEW

File: test/leeway-build.sh [hunk 1]
```diff
@@ -19,4 +19,5 @@ for COMPONENT in tests/components/*; do
     go test -trimpath -ldflags="-buildid= -w -s" -c -o bin/"$OUTPUT".test ./"$COMPONENT"
 done
 
-go test -trimpath -ldflags="-buildid= -w -s" -o bin/workspace -c ./tests/workspace
+echo building test tests/workspace
+go test -trimpath -ldflags="-buildid= -w -s" -o bin/workspace.test -c ./tests/workspace
```
Comment (utam0k) [inline test/leeway-build.sh:23]: It seems that the workspace test had not been run before, but I didn't know why.

File: test/tests/workspace/mount_proc_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+
+	agent "github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+)
+
+const (
+	numberOfMount = 500
+	parallel      = 5
```
Comment (utam0k) [inline test/tests/workspace/mount_proc_test.go:24]: There was no material to determine if it was a reasonable number. Do you have a good idea?
If it takes too long and becomes a problem, I'd consider adding an annotation for stress testing.

**Comment by utam0k** [comment]
wertf passed this integration test.
https://werft.gitpod-dev.com/job/gitpod-run-integration-tests-to-proc-test.12

File: test/leeway-build.sh (hunk 1 above)
Comment (utam0k) [reply test/leeway-build.sh:23]: ref: https://github.com/gitpod-io/gitpod/blob/30da76375c996109f243491b23e47feefab7217f/test/entrypoint.sh#L14

File: test/tests/workspace/mount_proc_test.go (hunk 2 above)
Comment (utam0k) [reply test/tests/workspace/mount_proc_test.go:24]: I was going to go with 1000 at first, but I'm having trouble finding any evidence that 1000 and 500 are different. The larger the number, the longer it takes to test.

Comment (Furisto) [reply test/tests/workspace/mount_proc_test.go:24]: Then lets go with 500 for now. If we see this problem again and the test does not show it, we can increase the number of iterations.

**Review by Furisto — APPROVED** [review]

File: test/tests/workspace/mount_proc_test.go (hunk 2 above)
Comment (utam0k) [reply test/tests/workspace/mount_proc_test.go:24]: 👍

Comment (utam0k) [reply test/tests/workspace/mount_proc_test.go:24]: @Furisto Thanks for your review!

**Pull request merged**